ui-tests/src/main/java/io/jenkins/plugins/coverage/CoverageSummary.java

```diff
@@ -0,0 +1,135 @@
+     * Creates a new page object representing the coverage summary on the build page of a job.
+     *
+     * @param parent
+     *         a finished build configured with a static analysis tool
+     * @param id
+     *         the type of the result page (e.g. simian, checkstyle, cpd, etc.)
+     */
+    public CoverageSummary(Build parent, String id) {
+        super(parent, parent.url(id));
+
+        this.summary = getElement(By.id(id + "-summary"));
+        this.coverageReportLink = getElement(By.id("coverage-hrefCoverageReport"));
+        this.referenceBuild = getElement(By.id("coverage-reference"));
+        this.coverage = this.summary.findElements(by.id("coverage-value"));
+        this.failMsg = getElement(By.id("coverage-fail-msg"));
+        this.coverageChanges = this.summary.findElements(by.id("coverage-change"));
+
+    }
+
+    public WebElement getCoverageReportLink() {
+        return coverageReportLink;
+    }
+
+    /**
+     * Opens attached {@link CoverageReport}.
+     * @return attached CoverageReport.
+     */
+    public CoverageReport openCoverageReport() {
+        return openPage(getCoverageReportLink(), CoverageReport.class);
+    }
+
+    /**
+     * Get coverage of Summary.
+     *
+     * @return Hashmap with coverage and value
+     */
+    public HashMap<String, Double> getCoverage() {
+        HashMap<String, Double> coverage = new HashMap<>();
+        for (WebElement result : this.coverage) {
+            String message = result.getText();
+            String type = message.substring(0, message.indexOf(":")).trim();
+            double value = Double.parseDouble(message.substring(message.indexOf(":") + 1, message.indexOf("%")).trim());
+            coverage.put(type, value);
+        }
+        return coverage;
+    }
+
+    /**
+     * Get relative changes of Coverage from previous build to this build.
+     *
+     * @return List of relative changes
+     */
+    public List<Double> getCoverageChanges() {
+        List<Double> changes = new ArrayList<>();
+        for (WebElement result : this.coverageChanges) {
+            String message = result.getText();
+            double value = Double.parseDouble(message.substring(1, message.indexOf("%")).trim());
+            changes.add(value);
+        }
+        return changes;
+    }
+
+    /**
+     * Get fail message in coverage summary.
+     *
+     * @return fail message
+     */
+    public String getFailMsg() {
+        return this.failMsg.getText();
+    }
+
+    /**
+     * Get url of reference build.
+     *
+     * @return url of reference build
+     */
+    public String getReferenceBuild() {
+        return this.referenceBuild.findElement(By.tagName("a")).getAttribute("href");
+    }
+
+    /**
+     * Open linked reference build in summary.
+     *
+     * @return {@link CoverageReport} of reference build
+     */
+    public CoverageReport openReferenceBuild() {
+        WebElement a = this.referenceBuild.findElement(By.tagName("a"));
```

Comment:
Extract `referenceBuild.findElement(By.tagName("a"))` into a method and reuse in line 104